Client/src/components/App.js

```diff
@@ -1,13 +1,55 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
+import Login from './contanier/Auth';
 import Mobile from './Layout/Mobile';
-// import Descktop from './Layout/Desktop';
-
-export default function App() {
-  return (
-    <div>
-      <Mobile />
-      {/* <Descktop /> */}
-    </div>
-  )
+import Descktop from './Layout/Desktop';
+import axios from 'axios';
+
+export default class App extends Component {
+  state = {
+    auth: false,
+    isLogin: false,
+    role: '',
+    id: '',
+  }
+  componentDidMount() {
+    axios
+      .get('/api/v1/auth')
+      .then(res => {
+        if (res.data.isAuth) {
+          this.setState({ auth: true, isLogin: true, id: res.data.id })
+        } else {
+          this.setState({ auth: true })
+        }
+      })
+  }
+
+  handleLogin = (id) => {
+    this.setState({ isLogin: true, id })
+  }
+
+  render() {
+    const { auth, isLogin, role, id } = this.state;
+    return (auth ?
+      <Router>
+        {isLogin ? (id.toString().length === 5 ?
```

Comment:
@AbdallahAmmar96 It's a little hard to follow all of the nested ternary operators in this render - I have a few ideas on how to rearrange this and make it easier to follow, if you'd like:
1) Move the `<h1>Loading</h1>` to the top, to short-circuit out of the render function early:
```
render() {
  const { auth, isLogin, role, id } = this.state;
  if (!auth) {
    return <h1>Loading..</h1>;
  }

  return (
    <Router>
    .....
```

2. Move the id.toString() check into a variable, and then conditionally include each route based on it:
```
const isStudent = id.toString().length === 5
...
{isStudent && <Route path="/student" component={Mobile} />}
{!isStudent && <Route path="/staff" component={Desktop} />}
...
```
Also, if the string length check could be changed into something more exact, like id === "(something goes here)", that would help.